### ISSUE TITLE
Feature Request: modify directory.lua script to use incoming call language

### ISSUE THREAD

**Issue by Touchedegris** [fusionpbx/fusionpbx#2686]
In a multi-tenant multi-language system, the directory TTS is only read in english.   There should be a condition to match the incoming call default language (ex: French company get French directory TTS, english company get english TTS and so on).

**Comment by markjcrane** [comment]
You can set default_language, default_voice and default_dialect. 
And then to be perfect it should use freeswitch phrases.

**Comment by Touchedegris** [comment]
Or there is an issue with the « if ».

My settings are:
default_language=fr
default_voice=june
default_dialect=ca

I am using the prompts from freeswitch.

De : francois bourassa
Envoyé : 13 juin 2017 08:50
À : fusionpbx/fusionpbx <reply@reply.github.com>; fusionpbx/fusionpbx <fusionpbx@noreply.github.com>
Cc : Author <author@noreply.github.com>
Objet : RE: [fusionpbx/fusionpbx] Feature Request: modify directory.lua script to use incoming call language (#2686)

The problem is with the name and extension spelling.  In the “directory.lua”, the lines  for “play the recorded name “, “annonce the first and last name” and “announce the extension number” are configure to use English language only.

Ex:
--announce the first and last names
if (speak_mod ~= nil and speak_voice ~= nil) then
                                                                                  session:execute("speak",speak_mod.."|"..speak_voice.."|"..row.first_name);
else
session:execute("say", "en name_spelled iterated "..row.first_name);
By modifying those lines for example”say”, “fr …” I can get the name and extension spelled in french, but it is system wide.  I think those lines should match the default_language variable from the incoming call/IVR menu and/or the user language if the prefix is dialed directly from an extension.


De : FusionPBX [mailto:notifications@github.com]
Envoyé : 13 juin 2017 01:48
À : fusionpbx/fusionpbx <fusionpbx@noreply.github.com<mailto:fusionpbx@noreply.github.com>>
Cc : Touchedegris <francois_bourassa@hotmail.com<mailto:francois_bourassa@hotmail.com>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Objet : Re: [fusionpbx/fusionpbx] Feature Request: modify directory.lua script to use incoming call language (#2686)


You can set default_language, default_voice and default_dialect.
And then to be perfect it should use freeswitch phrases.

**Comment by Touchedegris** [comment]
The problem is with the name and extension spelling.  In the “directory.lua”, the lines  for “play the recorded name “, “annonce the first and last name” and “announce the extension number” are configure to use English language only.

Ex:
--announce the first and last names
if (speak_mod ~= nil and speak_voice ~= nil) then
                                                                                  session:execute("speak",speak_mod.."|"..speak_voice.."|"..row.first_name);
else
session:execute("say", "en name_spelled iterated "..row.first_name);
By modifying those lines for example”say”, “fr …” I can get the name and extension spelled in french, but it is system wide.  I think those lines should match the default_language variable from the incoming call/IVR menu and/or the user language if the prefix is dialed directly from an extension.


De : FusionPBX [mailto:notifications@github.com]
Envoyé : 13 juin 2017 01:48
À : fusionpbx/fusionpbx <fusionpbx@noreply.github.com>
Cc : Touchedegris <francois_bourassa@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [fusionpbx/fusionpbx] Feature Request: modify directory.lua script to use incoming call language (#2686)


You can set default_language, default_voice and default_dialect.
And then to be perfect it should use freeswitch phrases.

**Comment by mafoo** [comment]
it may be better to use the playback application with phrases as this would like the language sort it out
I think the syntax would be session:execute("playback", "phrase:say_name:..row.first_name") however this would need us to maintain some more phrases into freeswitch

**Comment by Touchedegris** [comment]
Can the "en" be simply replaced by a variable?

**Comment by mafoo** [comment]
Not if you want to support all languages, some languages pronounce things in different order so to be more compatible we should use phrases as it lets the phrase deal with semantics of the language and we just say i want this phrase with this data
I don't have a good example I can give, I will see if I can make a sample tonight to show how it would get implemented.
I have already had to deal with this headache in another freeswitch project so its the way to go, however it is something of a overhaul to support properly and well, i would imagine this would be a 4.5 milestone but would be best to let mcrane call which way to go.
I could probably get the ground work working but I am going to need language contributors to get it working well in all languages

**Comment by Touchedegris** [comment]
Yeah I am facing an issue like this with the mailbox message reading in french.  However, since these lines only spells names and numbers, I don't think that the wording order matters on this one.  The rest of the directory is working fine, it's just the name and extension number spelling the issue.  Unless those lines are used for something else
